src/NCPkgStatusStrategy.cc

```diff
@@ -101,6 +101,15 @@ bool NCPkgStatusStrategy::setObjectStatus( ZyppStatus newstatus, ZyppSel slbPtr,
 {
     bool ok = false;
 
+    // workaround to clean previous state properly see bnc#916568 comment#8 especially
+    // basically problem is that if going from status taboo to status noinst sat solver
+    // is set to only remove lock and let solver to install it. What is actually expected
+    // by users is soft lock of package or pattern to not install it unless required
+    // so as workaround cycling thrue states, use different state as previous.
+    if (newstatus == S_NoInst)
+      slbPtr->setStatus( S_Protected );
```

Comment:
So this means that what we see in the installation will change
from `a+` (spc) `---` (spc) ` ` 
to `a+` (spc) `---` (spc) `-i-`?